pennylane_lightning/core/src/simulators/lightning_tensor/cutn/MPSCutn.hpp

```diff
@@ -0,0 +1,290 @@
+            Pennylane::LightningGPU::Util::complexToCu<ComplexT>({1.0, 0.0});
+
+        for (size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            tensors_[i].getDataBuffer().zeroInit();
+            size_t target = 0;
+            size_t idx = BaseType::getNumQubits() - size_t{1} - i;
+
+            // Rightmost site
+            if (i == 0) {
+                target = basisState[idx];
+            } else {
+                target = basisState[idx] == 0 ? 0 : maxBondDim_;
+            }
+
+            PL_CUDA_IS_SUCCESS(
+                cudaMemcpy(&tensors_[i].getDataBuffer().getData()[target],
+                           &value_cu, sizeof(CFP_t), cudaMemcpyHostToDevice));
+        }
+
+        updateQuantumStateMPS_(getSitesExtentsPtr().data(),
+                               getTensorsDataPtr().data());
+    };
+
+    /**
+     * @brief Get the full state vector representation of MPS quantum state.
+     *
+     * NOTE: This API is only for MPS unit tests purpose only, given that the
+     * full statevector system is small. Attempt to apply this method to largr
+     * systems will lead to memory issue.
+     *
+     * @return Full state vector representation of MPS quantum state on host
+     * std::vector<ComplexT>
+     */
+    auto getDataVector() -> std::vector<ComplexT> {
+        PL_ABORT_IF_NOT(this->MPSFinalized_ == false,
+                        "getDataVector() method to return the full state "
+                        "vector can't be called "
+                        "after cutensornetStateFinalizeMPS is called");
+
+        this->MPSFinalized_ = true;
+
+        // 1D representation
+        std::vector<size_t> output_modes(size_t{1}, size_t{1});
+        std::vector<size_t> output_extent(
+            size_t{1}, size_t{1} << BaseType::getNumQubits());
+        CudaTensor<Precision> output_tensor(output_modes.size(), output_modes,
+                                            output_extent,
+                                            BaseType::getDevTag());
+
+        std::vector<void *> output_tensorPtr(
+            size_t{1},
+            static_cast<void *>(output_tensor.getDataBuffer().getData()));
+
+        std::vector<int64_t *> output_extentsPtr;
+        std::vector<int64_t> extent_int64(
+            size_t{1},
+            static_cast<int64_t>(size_t{1} << BaseType::getNumQubits()));
+        output_extentsPtr.emplace_back(extent_int64.data());
+
+        BaseType::computeState(output_extentsPtr, output_tensorPtr);
+
+        std::vector<ComplexT> results(output_extent.front());
+        output_tensor.CopyGpuDataToHost(results.data(), results.size());
+
+        return results;
+    }
+
+  private:
+    /**
+     * @brief The helper function for constuctor.
+     */
+    void initHelper_() {
+        // Configure extents for each sites
+        for (size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            std::vector<size_t> localSiteModes;
+            std::vector<size_t> localSiteExtents;
```

Comment:
Is there a change we can make to avoid creating the vectors here, then fully copying them across later? Maybe moving them in somehow?

Reply:
use `std::move` instead.